scripts/normalize/lib/docs1-link-rewriter.js

```diff
@@ -0,0 +1,5204 @@
+        linkStats.nameNotMapped[title] = (linkStats.nameNotMapped[title]||0)+1
+        linkStats.unmappedRootLinks++;
+      }
+    }
+    else if (deepPath.test(pathname))
+    {
+      const latestVersionedPath = /^(\/edb-docs\/[pd]\/[^\/]+\/)(?:[^\/]+\/)*latest(?:\/[^\/]+)*\/?$/;
+      const unversionedPath = /^\/edb-docs\/[pd]\/[^\/]+\/?$/;
+      const semverSegment = /\/(\d+(?:\.\d+)*)(\/|$)/;
+      newPath = urlMap[pathname];
+      if (!newPath && unversionedPath.test(pathname))
+      {
+        newPath = urlMap[pathname.replace(/\/?$/, '/latest')]
+        // would be nice if there was always a latest, but... If not, try to dig up the top version
+        if (!newPath)
+        {
+          const versions = Object.keys(urlMap).filter(p => p.startsWith(pathname) && semverSegment.test(p)).map(p => p.match(semverSegment)[1]);
+          newPath = urlMap[pathname.replace(/\/?$/, '/' + versions.sort(versionComp).pop())];
+        }
+      }
+      // contrasting problem: path with latest in it, but no mapping
+      if (latestVersionedPath.test(pathname))
+      {
+        const versions = Object.keys(urlMap).filter(p => p.startsWith(pathname.match(latestVersionedPath)[1]) && semverSegment.test(p)).map(p => p.match(semverSegment)[1]);
+        newPath = urlMap[pathname.replace('/latest', '/' + versions.sort(versionComp).pop())]
+      }
+
+      // handle multi-hop redirects
+      // also handle the possibility of an infinite redirect loop
+      let hops = 0;
+      while (urlMap[newPath] && hops++ < 10) newPath = urlMap[newPath];
+      if (hops==10)
+        console.error(`TOO MANY redirects for ${newPath}`);
+
+      if (!newPath)
+        linkStats.pathNotMapped.push(pathname);
+      else
+        linkStats.remappedDeepLinks++;
+    }
+    else
+    {
+      console.error(`Donno what to do with '${url}'`);
+    }
+
+
+    if (newPath)
+    {
+      text.replace(url.href, docs2Url + newPath);
+    }
+
+    return {newPath, text};
+  }
+
+  function getContext(node, ancestors, lookbackN)
+  {
+    // first try just looking at previous sublings
+    const parent = ancestors[ancestors.length-1];
+    const index = parent.children.indexOf(node);
+    const textSiblings = parent.children.slice(0, index);
+    const result = [];
+    lookbackN = Math.min(lookbackN || textSiblings.length, textSiblings.length);
+    for (let i=textSiblings.length; lookbackN && i>=0; --i)
+    {
+      const segment = mdast2string({children: [textSiblings[i]]}).trim();
+      if (segment.length)
+      {
+        result.unshift(segment);
+        lookbackN--;
+      }
+    }
+    if (result.length) return result.join(' ');
+
+    // nothing there? Go up a level and do the same thing, but only look back one node
+    if (ancestors.length > 1)
+      return getContext(parent, ancestors.slice(0, -1), 1);
+    return "";
+  }
+}
+
+const titleMap = {
```

Comment:
I should have thought to send you my redirects metadata file, it has this information in a fairly similar format  😬